.github/workflows/tests-pr.yml

```diff
@@ -1,8 +1,9 @@
+name: Tests Pull Requests
+concurrency:
+  group: tests-${{ github.ref }}
+  cancel-in-progress: true
 on:
-  push:
-    branches: [main]
```

Comment:
There was no (nice) way to have the concurrency only apply to non-main branches, so moved main branch lint and test over to the main branch workflow.